src/malli/transform.cljc

```diff
@@ -95,6 +96,15 @@
       x)
     x))
 
+#?(:clj
+   (defn -string->uri [x]
+     (if (string? x)
+       (try
+         (URI. x)
+         (catch Exception _
```

Comment:
Would it be better to catch `java.net.URISyntaxException` here?

Reply:
Your call. The body of the `try` is so small there's little possibility of masking unrelated exceptions.

Reply:
You're right that the body is small, but I prefer `URISyntaxException` for explicitness - so I have updated it.

Reply:
Hmm, that fails the Babashka job, so I guess I change it back :-)

https://github.com/metosin/malli/actions/runs/6493910884/job/17635777011

Reply:
I couldn't get the recommended macro approach to work, so I just put a comment in to tighten up the catch clause once our version of babashka has the required commit.